src/sys/sendfile.rs

```diff
@@ -2,12 +2,83 @@ use std::os::unix::io::RawFd;
 use std::ptr;
 
 use libc::{self, off_t};
+#[cfg(target_os = "freebsd")]
+use libc::c_int;
 
 use Result;
+#[cfg(target_os = "freebsd")]
+use sys::uio::IoVec;
 use errno::Errno;
 
+#[cfg(any(target_os = "linux",
+          target_os = "android",
+          all(unix, target_env = "uclibc")))]
 pub fn sendfile(out_fd: RawFd, in_fd: RawFd, offset: Option<&mut off_t>, count: usize) -> Result<usize> {
     let offset = offset.map(|offset| offset as *mut _).unwrap_or(ptr::null_mut());
     let ret = unsafe { libc::sendfile(out_fd, in_fd, offset, count) };
     Errno::result(ret).map(|r| r as usize)
 }
+
+#[cfg(any(target_os = "dragonfly",
+          target_os = "freebsd",
+          target_os = "ios",
+          target_os = "macos"))]
+#[allow(missing_debug_implementations)]
+#[allow(missing_copy_implementations)]
+#[repr(C)]
+pub struct SfHdTr(libc::sf_hdtr);
```

Comment:
I'd love to see a better name for this struct compared to the cryptic shorthand libc uses. `SendfileHeader` I think would work

Reply:
How about `SendfileHeaderTrailer`?